xray/conventions.py

```diff
@@ -720,9 +727,13 @@ def decode_cf_variable(var, concat_characters=True, mask_and_scale=True,
             attributes['_FillValue'] = attributes.pop('missing_value')
 
         fill_value = pop_to(attributes, encoding, '_FillValue')
+        if isinstance(fill_value, np.ndarray) and len(fill_value) != 1:
```

Comment:
could you add a test that the warning is raised? e.g., see 
https://github.com/xray/xray/blob/f8d79819f83cc23c9e587847bb683b11b0d307b8/xray/test/test_conventions.py#L312-L318